### PR TITLE
porting conda plugin from v1 to v2 so we can use it in core20

### PR DESCRIPTION
**Pull request by ycheng**
- [V] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [V] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?
-----

For CLA, don't know which contact / PM I should fill in.
For test/unit, still struggling to know what exactly failed. Need more time to check what's going on.
  I saw lots of "AssertionError: Unhandled command: ['file', '-b', '/usr/bin/python3.8']", got some hinit?
Thanks.

### CODE REVIEW

File: snapcraft/plugins/v2/conda.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        )
+
+    def _get_install_env_command(self) -> str:
+        cmd = ["${HOME}/miniconda.sh", "-bfp", "${HOME}/miniconda"]
+        return " ".join(cmd)
+
+    def _get_deploy_command(self) -> str:
+        import pprint as pp
+
+        pp.pprint(self.options)
```
Comment (sergiusens) [inline snapcraft/plugins/v2/conda.py:98]: I suppose this is a left over from trouble shooting?

File: snapcraft/plugins/v2/conda.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        conda_target_prefix = os.path.join(
```
Comment (sergiusens) [inline snapcraft/plugins/v2/conda.py:99]: can you use pathlib.Path instead

**Review by sergiusens — CHANGES_REQUESTED** [review]
Thanks for this contribution.

I've added a few inline comments and here are two more that need addressing
- tests/unit/plugins/v2/test_conda.py for unit tests needs to be developed
- tests/spread/plugins/v2 for conda needs to exist. If this project can rebuild on the fly, it would be good to piggy back on that fixutre

File: snapcraft/plugins/v2/conda.py (hunk 1 above)
Comment (ycheng) [reply snapcraft/plugins/v2/conda.py]: removed.

File: snapcraft/plugins/v2/conda.py (hunk 2 above)
Comment (ycheng) [reply snapcraft/plugins/v2/conda.py]: Maybe just a Linux-style path would be fine.

**Comment by ycheng** [comment]
IMHO this can be merged.

I have the plan to add another two option:
conda-package-file
  You can put channel, package, and version spec.
conda-create-params
  Command-line parameters if you have extra customise requirements.

For me, I'd like to use the conda-package-file that makes it easier to use some existing package-file per my roadmap. Options conda-create-params is more for completeness.

I think those can be the next iteration because I am not so sure if I do get the bandwidth to work on it in sort time.

**Review by cjp256 — APPROVED** [review]
usingI would suggest considering using $SNAPCRAFT_PART_BUILD instead of $HOME to keep things isolated, but otherwise LGTM.  And thanks for making me think about the use of CONDA_TARGET_PREFIX_OVERRIDE, relevant to something else I was looking at :D

**Review by sergiusens — APPROVED** [review]

File: snapcraft/plugins/v2/conda.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+"""
+
+from textwrap import dedent
+from typing import Any, Dict, List, Set
+
+from snapcraft.plugins.v2 import PluginV2
+
+
+def _get_miniconda_source(version: str) -> str:
+    """Return tuuple of source_url and source_checksum (if known)."""
```
Comment (sergiusens) [inline snapcraft/plugins/v2/conda.py:45]:
```suggestion
    """Return tuple of source_url and source_checksum (if known)."""
```

File: snapcraft/plugins/v2/conda.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+
+    - conda-packages
+      (list of strings, default [])
+      conda packages to install
```
Comment (sergiusens) [inline snapcraft/plugins/v2/conda.py:27]:
```suggestion
      List of conda packages to install.
```

File: snapcraft/plugins/v2/conda.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+    - conda-python-version
+      string
+      python version like 3.8, etc.
```
Comment (sergiusens) [inline snapcraft/plugins/v2/conda.py:31]:
```suggestion
      Python version major and minor version (e.g. 3.8).
```

File: snapcraft/plugins/v2/conda.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+    - conda-miniconda-version
+      string, default latest
+      the miniconda to initialize.
```
Comment (sergiusens) [inline snapcraft/plugins/v2/conda.py:35]:
```suggestion
      The version of miniconda to initialize.
```

**Review by sergiusens — APPROVED** [review]

**Comment by ycheng** [comment]
@sergiusens thank you!

**Pull request merged**